### ISSUE TITLE
[Bug] Calling a public inline function that uses an internal constant triggers an ICE

### ISSUE THREAD

**Issue by vineethk** [aptos-labs/aptos-core#11022]
# 🐛 Bug

Let's say we have a public inline function in one module which refers to a `const` in its own module. If a function in another module calls this inline function, then `aptos move compile` triggers an ICE.

## To reproduce

Run `aptos move compile` on the following code:

**Code snippet to reproduce**
```
module 0xc0ffee::dummy1 {
    const CC: u64 = 1;

    public inline fun expose(): u64 {
        CC
    }
}

module 0xc0ffee::dummy2 {
    public fun main(): u64 {
        0xc0ffee::dummy1::expose()
    }
}
```

**Stack trace/error message**
```
bug[ICE12001]: BYTECODE GENERATION FAILED
  │
9 │ module 0xc0ffee::dummy2 {
  │                  ^^^^^^ IR ERROR: Missing constant definition for CC```
```

## Expected Behavior

A compiler error indicating what the actual problem is, instead of an ICE and IR error.

## System information

**Please complete the following information:**
- Aptos Core Version: current main branch

**Comment by brmataptos** [comment]
I've added this test case to the move-compiler-v2 inlining PR.  Compiler V2 currently just inlines the constant value (not the constant name) in the inlining case, so it will run with no problem.  Access checks are working in other cases, as Compiler V2 seems to successfully generate an error for a function call.  Is this acceptable?
@wrwg.

Hmm.  Maybe we need a test for accessing a non-local constant in V2.  Did you test that as apart of the access checking, @vineethk .

**Comment by vineethk** [comment]
> Hmm. Maybe we need a test for accessing a non-local constant in V2. Did you test that as apart of the access checking, @vineethk .

Yes: https://github.com/aptos-labs/aptos-core/blob/main/third_party/move/move-compiler-v2/tests/visibility-checker/constant_use_across_mod.move

**Comment by vineethk** [comment]
Fixed: #15785
